cmd/influx/task.go

```diff
@@ -3,342 +3,425 @@ package main
 import (
 	"context"
 	"fmt"
-	"io"
-	"os"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/tenant"
+
 	"github.com/influxdata/influxdb/v2"
-	"github.com/influxdata/influxdb/v2/cmd/influx/internal"
 	"github.com/influxdata/influxdb/v2/http"
 	"github.com/spf13/cobra"
 )
 
+type taskSVCsFn func() (influxdb.TaskService, influxdb.OrganizationService, error)
+
+func newTaskSVCs() (influxdb.TaskService, influxdb.OrganizationService, error) {
+	httpClient, err := newHTTPClient()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	orgSvc := &tenant.OrgClientService{Client: httpClient}
+	return &http.TaskService{Client: httpClient}, orgSvc, nil
+}
+
 func cmdTask(f *globalFlags, opt genericCLIOpts) *cobra.Command {
-	runE := func(cmd *cobra.Command, args []string) error {
-		seeHelp(cmd, args)
-		return nil
+	builder := newCmdTaskBuilder(newTaskSVCs, f, opt)
+	return builder.cmd()
+}
+
+type cmdTaskBuilder struct {
+	opts        genericCLIOpts
+	globalFlags *globalFlags
+
+	svcFn taskSVCsFn
+
+	taskID         string
+	runID          string
+	taskPrintFlags taskPrintFlags
+	// todo: fields of these flags structs could be pulled out for a more streamlined builder struct
+	taskCreateFlags      taskCreateFlags
+	taskFindFlags        taskFindFlags
+	taskRerunFailedFlags taskRerunFailedFlags
+	taskUpdateFlags      taskUpdateFlags
+	taskRunFindFlags     taskRunFindFlags
+	org                  organization
+}
+
+func newCmdTaskBuilder(svcsFn taskSVCsFn, f *globalFlags, opts genericCLIOpts) *cmdTaskBuilder {
+	return &cmdTaskBuilder{
+		globalFlags: f,
+		opts:        opts,
+		svcFn:       svcsFn,
 	}
+}
 
-	cmd := opt.newCmd("task", runE, false)
+func (t *cmdTaskBuilder) cmd() *cobra.Command {
+	cmd := t.newCmd("task", nil)
```

Comment:
I can only point out where I mirrored this from since I don't know too much myself. 
https://github.com/influxdata/influxdb/blob/e77c4670d6c6aaab84f43b69fb5e2f3bbb3be910/cmd/influx/bucket.go#L43-L44